cohort-engine-api-web/src/main/java/com/ibm/cohort/engine/api/service/CohortEngineRestHandler.java

```diff
@@ -279,5 +287,62 @@ public Response getMeasureParametersById(@Context HttpHeaders httpHeaders,
 		}
 	}
 
+	//todo move this to the top (keeping here for locality in development)
+	public static String VALUE_SET_PART = "VALUE_SET";
+
+	@POST
+	@Path("/valueset/")
+	@Consumes(MediaType.MULTIPART_FORM_DATA)
+	@Produces({ MediaType.APPLICATION_JSON })
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Successful Operation", response = String.class),
+			//todo new response code for already exists
+			@ApiResponse(code = 400, message = "Bad Request", response = ServiceErrorList.class),
+			@ApiResponse(code = 500, message = "Server Error", response = ServiceErrorList.class)
+	})
+	@ApiOperation(value = "Insert a new value set to the fhir server or, if it already exists, update it in place", authorizations = {@Authorization(value = "BasicAuth") })
+	public Response putValueSet(@Context HttpHeaders httpHeaders,
+								@ApiParam(value = ServiceBaseConstants.MINOR_VERSION_DESCRIPTION, required = true, defaultValue = ServiceBuildConstants.DATE) @QueryParam("version") String version,
+								@ApiParam(value = CohortEngineRestHandler.FHIR_ENDPOINT_DESC, required = true, defaultValue = CohortEngineRestHandler.DEFAULT_FHIR_URL) @QueryParam("fhir_server_rest_endpoint") String fhirEndpoint,
+								@ApiParam(value = CohortEngineRestHandler.FHIR_TENANT_ID_DESC, required = true, defaultValue = "default") @QueryParam("fhir_server_tenant_id") String fhirTenantId,
+								@ApiParam(value = CohortEngineRestHandler.FHIR_TENANT_HEADER_DESC, defaultValue = IBMFhirServerConfig.DEFAULT_TENANT_ID_HEADER) @QueryParam("fhir_server_tenant_id_header") String fhirTenantIdHeader,
+								@ApiParam(value = CohortEngineRestHandler.FHIR_DS_HEADER_DESC, defaultValue = IBMFhirServerConfig.DEFAULT_DATASOURCE_ID_HEADER) @QueryParam("fhir_data_source_id_header") String fhirDataSourceIdHeader,
+								@ApiParam(value = CohortEngineRestHandler.FHIR_DS_ID_DESC) @QueryParam("fhir_data_source_id") String fhirDataSourceId,
+								//todo default to false
+								@QueryParam("updateIfExists") boolean updateIfExists,
+								//todo can assume only need one file at a time
+								@ApiParam(value = "Multipart form request containing value set spreadsheets") IMultipartBody multipartBody
```

Comment:
If we wanted to support multiple files at a time, it would be pretty easy to loop through the attachments on the multipart request and check the Content-Type as .xlsx or something like that. It wouldn't need an additional API parameter.

Reply:
I'm fine to do that, though the feedback I've gotten right now is that there's not going to be a use case for the UI to upload multiple value sets at once.